src/dropdown.ts

```diff
@@ -48,54 +51,81 @@ export class FdsDropdownEvent<T> extends CustomEvent<FdsDropdownOption<T>> {
+  @property({type: Boolean}) required: boolean = false
+  @property() name?: string
 
   override firstUpdated(): void {
     this.tabIndex = 0
+    this.setValidity();
   }
 
   override render(): TemplateResult {
-    const optionsList = html`
-      <div part="options-list" class="options-list">
-        ${this.options.map(
-          option => html`
-            <div
+    const singleOptionItem = (option) => html`
+            <li
               @click=${(): void => this.handleSelect(option)}
               @keypress=${(e: KeyboardEvent): void => this.handleKeypress(e, option)}
               class=${`ui-label-text option ${this.getOptionCssClass(option)}`}
               tabindex=${0}
               aria-selected=${this.value === option}
             >
               ${this.getLabel(option)}
-            </div>
+            </li>
           `
+    const multipleOptionItem = (option) => html`
+            <li>
+                <label class="ui-label-text option option-multiple ${this.getOptionCssClass(option)}">
+                    <input type="checkbox"
+                           @change="${(): void => this.handleMultiSelect(option)}"
+                           ${this.name ? html`name="${this.name}"`:""}
+                           ${option.value ? html`value="${option.value}"`:""}>
+                    ${this.getLabel(option)}
+                </label>
+            </li>
+        `
+    const optionsList = html`
+      <ul part="options-list" id="options-list" role="listbox" aria-label="options" class="options-list" aria-multiselectable="true">
+        ${this.options.map(
+          option => this.multiple ? multipleOptionItem(option) : singleOptionItem(option)
         )}
-      </div>
+      </ul>
     `
+    const isFirstRender = this.renderRoot.children.length === 0
 
     return html`
-      <button
-        @click=${(): boolean => (this._open = !this._open)}
-        ?disabled=${this.disabled}
-        class=${`ui-label-text ${this.getButtonCssClass()}`}
-        aria-haspopup=${true}
-        aria-expanded=${this._open}
-      >
-        <div>${this.getLabel(this.value) ?? this.placeholder}</div>
-        <fds-icon .icon=${this._open ? 'chevron-up' : 'chevron-down'}></fds-icon>
-      </button>
-      ${this._open ? optionsList : null}
+      <div class="dropdown-wrapper">
+        <button
+          @click=${(): void => {
+            const buttonElement = this.getButton();
+            buttonElement.ariaExpanded = (!(buttonElement.ariaExpanded === 'true')).toString()
+          }}
+          ?disabled=${this.disabled}
+          class=${`ui-label-text ${this.getButtonCssClass()}`}
+          role="combobox"
+          aria-controls="options-list"
+          aria-expanded=${isFirstRender ? "false" : this.getButton().ariaExpanded}
+        >
+          <div>${this.getLabel(this.value) ?? this.placeholder}</div>
+          <fds-icon icon='chevron-up'></fds-icon>
+          <fds-icon icon='chevron-down'></fds-icon>
```

Comment:
Mikäli aria-expanded käyttö on tässä perusteltua, niin voisiko tuosta ylempää ottaa tuon 
`isFirstRender ? "false" : this.getButton().ariaExpanded` ehdon tuloksen kokonaisuudessaan johonkin muuttujaan (esim. `ariaExpanded`), ja tehdä näin:
```suggestion
          <fds-icon .icon=${ariaExpanded ? 'chevron-up' : 'chevron-down'}></fds-icon>
```
Ehkä vain tottumus kysymys, mutta vähän erikoinen ensivilkaisulla kun molemmat ikonit liitetään domiin ja niiden näkyvyyttä vain kontrolloidaan css:llä.

Reply:
Tuli kokeiltuakin tuommoista alkuunsa. Ei vain tuota `render` funktiota ajettu, kun vaihto `ariaExpanded`in arvoa JavaScritillä, jolloinka tuo ikoni ei päivittynyt. Voisin vielä kokeilla saako tuon jotenkin tunkattua tuohon suuntaan vielä.

Reply:
Niin just, kun toi ariaExpanded ei ole reaktiivinen arvo, niin se ei päivitä tuota komponenttia. No voidaan mun puolesta mennä tällä mikäli et jotain selkeämpää ratkaisua vielä satu keksimään.